libraries/tock-cells/src/optional_cell.rs

```diff
@@ -55,38 +51,54 @@ impl<T: Copy> OptionalCell<T> {
 
     /// Check if the cell contains something.
     pub fn is_some(&self) -> bool {
-        self.value.get().is_some()
+        let value = self.value.take();
+        let out = value.is_some();
+        self.value.set(value);
+        out
     }
 
     /// Check if the cell is None.
     pub fn is_none(&self) -> bool {
-        self.value.get().is_none()
+        let value = self.value.take();
+        let out = value.is_none();
+        self.value.set(value);
+        out
     }
 
     /// Returns true if the option is a Some value containing the given value.
     pub fn contains(&self, x: &T) -> bool
     where
         T: PartialEq,
     {
-        self.value.get().contains(x)
+        let value = self.value.take();
+        let out = value.contains(x);
+        self.value.set(value);
+        out
     }
 
     /// Returns the contained value or panics if contents is `None`.
-    pub fn expect(&self, msg: &str) -> T {
+    pub fn expect(&self, msg: &str) -> T
+    where
+        T: Copy,
+    {
```

Comment:
This still seems to require the contained type is `Copy`?

Reply:
Several methods of `OptionalCell` -- including `expect` -- require `T: Copy`.

Unless it panics, `expect` returns a copy of the value contained in the `OptionalCell` without modifying the `OptionalCell`, which definitely requires `T: Copy`.

Reply:
This makes sense, I was just confused by the first sentence of the PR "This pull requests makes OptionalCell work with non-Copy types"